crates/rust-analyzer/src/config.rs

```diff
@@ -196,110 +175,113 @@ config_data! {
+        /// Enables highlighting of related references while hovering your mouse above any identifier.
+        highlightRelated_references_enable: bool = "true",
         /// Enables highlighting of all break points for a loop or block context while hovering your mouse above any `async` or `await` keywords.
-        highlightRelated_yieldPoints: bool = "true",
-
-        /// Use semantic tokens for strings.
-        ///
-        /// In some editors (e.g. vscode) semantic tokens override other highlighting grammars.
-        /// By disabling semantic tokens for strings, other grammars can be used to highlight
-        /// their contents.
-        highlighting_strings: bool = "true",
-
-        /// Whether to show documentation on hover.
-        hover_documentation: bool       = "true",
-        /// Use markdown syntax for links in hover.
-        hover_linksInHover |
-        hoverActions_linksInHover: bool = "true",
+        highlightRelated_yieldPoints_enable: bool = "true",
 
         /// Whether to show `Debug` action. Only applies when
         /// `#rust-analyzer.hoverActions.enable#` is set.
-        hoverActions_debug: bool           = "true",
+        hover_actions_debug_enable: bool           = "true",
         /// Whether to show HoverActions in Rust files.
-        hoverActions_enable: bool          = "true",
+        hover_actions_enable: bool          = "true",
         /// Whether to show `Go to Type Definition` action. Only applies when
         /// `#rust-analyzer.hoverActions.enable#` is set.
-        hoverActions_gotoTypeDef: bool     = "true",
+        hover_actions_gotoTypeDef_enable: bool     = "true",
         /// Whether to show `Implementations` action. Only applies when
         /// `#rust-analyzer.hoverActions.enable#` is set.
-        hoverActions_implementations: bool = "true",
+        hover_actions_implementations_enable: bool = "true",
         /// Whether to show `References` action. Only applies when
         /// `#rust-analyzer.hoverActions.enable#` is set.
-        hoverActions_references: bool      = "false",
+        hover_actions_references_enable: bool      = "false",
         /// Whether to show `Run` action. Only applies when
         /// `#rust-analyzer.hoverActions.enable#` is set.
-        hoverActions_run: bool             = "true",
+        hover_actions_run_enable: bool             = "true",
+
+        /// Whether to show documentation on hover.
+        hover_documentation_enable: bool       = "true",
+        /// Use markdown syntax for links in hover.
+        hover_links_enable: bool = "true",
+
+        /// Whether to enforce the import granularity setting for all files. If set to false rust-analyzer will try to keep import styles consistent per file.
+        imports_enforceGranularity: bool              = "false",
+        /// How imports should be grouped into use statements.
+        imports_granularity: ImportGranularityDef  = "\"crate\"",
```

Comment:
I'd like these two to be both under `imports_granularity` but I can't think of a fiting name for the latter for that

Reply:
Ye it guesses, then falls back to granularity if the guess permits it. `default` doesn't quite fit though I think, since its only a default when guessing is allowed, otherwise its definitive.